poetry/utils/env.py

```diff
@@ -1578,17 +1580,19 @@ def __init__(
     def get_pip_command(self, embedded: bool = False) -> List[str]:
         return [self._bin("python"), self.pip_embedded if embedded else self.pip]
 
-    def _run(self, cmd: List[str], **kwargs: Any) -> int:
+    def _run(self, cmd: List[str], **kwargs: Any) -> Optional[int]:
```

Comment:
this change, while correct, actually increases the total number of errors, since the `None` case is not handled in the calling code.